File: math-and-number-theory/3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        copy = n
        ds = 0
        dp = 1
        while n>0:
            n,digit= divmod(n,10)
            ds+=digit
            dp*=digit
        return copy %(ds+dp) == 0
        
        

File: math-and-number-theory/3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int x = n;
        int digit_product= 1,digit_sum = 0;
        while(n>0){
            int rem = n%10;
            digit_sum +=(rem);
            digit_product *= rem;
            n/=10;
        }
        return x%(digit_sum+digit_product)==0? true: false;
    }
}